### PR TITLE
Update go-rsyslog-pstats to support custom statsd host and prefix with hostname

### PR DESCRIPTION
**Pull request by justinpersaud**
I had to make a few changes to the original code for our needs

- support passing in host param as statsd daemon
- prefix metrics with hostname otherwise we don't know which server is sending the metric

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -127,7 +128,12 @@ func findNums(prefix string, kvs map[string]interface{}, out io.Writer) {
 			continue
 		}
 
-		_, err := fmt.Fprintf(out, "rsyslog.%v.%v:%d|g\n", prefix, sanitizeKey(k), int(vf))
+		hostname, err := os.Hostname()
```
Comment (justinpersaud) [inline main.go:132]: I have not tested this yet but I believe this is the right place to insert the hostname prefix. I considered doing this via the statsd input in Telegraf configuration, but that would interfere with our bedrock logs to statsd script.

File: main.go [hunk 2]
```diff
@@ -127,7 +128,12 @@ func findNums(prefix string, kvs map[string]interface{}, out io.Writer) {
...
+		if err != nil {
+			el.Println("Error determining hostname", err)
+		}
+
+		_, err = fmt.Fprintf(out, hostname+".rsyslog.%v.%v:%d|g\n", prefix, sanitizeKey(k), int(vf))
```
Comment (justinpersaud) [inline main.go]: thinking about this a bit, but I wonder if I need to modify the hostname

`os.Hostname()` should return something like `db1.sjc`, but that `.` will cause issues and needs to be converted to `-`, right?

Comment (rafecolton) [inline main.go]: NAB you can format like this

```go
		if _, err = fmt.Fprintf(out, hostname+".rsyslog.%v.%v:%d|g\n", prefix, sanitizeKey(k), int(vf)); err != nill {
```

Comment (rafecolton) [reply main.go]: Yeah the `.` will cause another level of nesting, so you will indeed need to replace it with a `-`

**Review by rafecolton — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (rafecolton) [reply main.go]: Come to think of it, I would put `rsyslog` first and then the hostname so they are easier to find by drilling down

Comment (justinpersaud) [reply main.go]: I am thinking to leave it because it doesn't follow the conventions errors are handled in the rest of the file.

**Comment by justinpersaud** [comment]
updated

File: main.go [hunk 3]
```diff
@@ -133,7 +134,10 @@ func findNums(prefix string, kvs map[string]interface{}, out io.Writer) {
 			el.Println("Error determining hostname", err)
 		}
 
-		_, err = fmt.Fprintf(out, hostname+".rsyslog.%v.%v:%d|g\n", prefix, sanitizeKey(k), int(vf))
+		// Replace periods with hyphens for graphite compatbility
+		hostnameFormatted := strings.Replace(hostname, ".", "-", -1)
```
Comment (rafecolton) [inline main.go]: NAB: You can use `ReplaceAll` and then skip the -1

```suggestion
		hostnameFormatted := strings.ReplaceAll(hostname, ".", "-")
```

**Review by rafecolton — APPROVED** [review]

**Comment by justinpersaud** [comment]
ok, looks like we're all good here and I can test this out more tomorrow.

I think because I forked this here I'm getting tests from upstream -- do you agree we just ignore and merge?

They likely won't pass because I haven't updated the test cases to account for my hostname change and the go version they used to build it originally is likely an older one, so I'm not sure it's worth the time investment to fix all of those.

**Review by rafecolton — APPROVED** [review]
> do you agree we just ignore and merge

Yeah, I agree we can skip this for now. I think we should fix it with the next PR though if we change too much more

**Pull request merged**